### PR TITLE
Refactor: join display modules in package

### PR DESCRIPTION
**Pull request by penguinolog**
* Simplify support, when amount of code will raise
* `RealTerminal.tty_signal_keys`: make windows-specific branch
* Do not deprecate: `curses_display` and `raw_display` imports

##### Checklist
- [x] I've ensured that similar functionality has not already been implemented
- [X] I've ensured that similar functionality has not earlier been proposed and declined
- [X] I've branched off the `master` or `python-dual-support` branch
- [X] I've merged fresh upstream into my branch recently
- [X] I've ran `tox` successfully in local environment
- [X] I've included docstrings and/or documentation and/or examples for my code (if this is a new feature)

Partial #447

### CODE REVIEW

File: urwid/lcd_display.py
Comment (penguinolog) [inline urwid/lcd_display.py:1]: Special case, can be prepared for deprecation (no deprecation at this moment)

File: urwid/html_fragment.py
Comment (penguinolog) [inline urwid/html_fragment.py:1]: Special case, can be prepared for deprecation (no deprecation at this moment)

File: urwid/escape.py
Comment (penguinolog) [inline urwid/escape.py:1]: Was used internally, old location deprecated

File: urwid/display_common.py
Comment (penguinolog) [inline urwid/display_common.py:1]: Base class and helpers, directly used internally.
Main public code is accessed from package namespace without changes.

File: urwid/__init__.py [hunk 1]
```diff
@@ -194,10 +194,16 @@
...
 )
 
-from . import event_loop, widget
+from . import display, event_loop, widget
 
 from urwid.treetools import ParentNode, TreeListBox, TreeNode, TreeWalker, TreeWidget, TreeWidgetError  # isort: skip
 
+# Optional displays with external dependencies
+try:
+    from .display import curses as curses_display  # backward compatibility
```
Comment (penguinolog) [inline urwid/__init__.py:203]: this way we have full backward compatibility without extra files.
Even `PendingDeprecation` is too much for such common module at this moment

**Comment by wardi** [comment]
-1 from me. This is a small library, I don't think we need more namespaces. Also things like `urwid.display.curses` just will shadow the core `curses` module, right?

File: urwid/display/web.py [hunk 2]
```diff
@@ -594,8 +119,9 @@
 ]
 
 
-class Screen:
+class Screen(BaseScreen):
```
Comment (penguinolog) [inline urwid/display/web.py:122]: Subclass base class to warranty interfaces

File: urwid/display/web.py [hunk 3]
```diff
@@ -722,17 +256,6 @@ def stop(self):
     def set_input_timeouts(self, *args):
         pass
 
-    def run_wrapper(self, fn):
```
Comment (penguinolog) [inline urwid/display/web.py:725]: already in base class implemented correct way

**Comment by penguinolog** [comment]
> -1 from me. This is a small library, I don't think we need more namespaces.
After "clean-up" we have a bit too many files for root module.

> Also things like `urwid.display.curses` just will shadow the core `curses` module, right?
No, if nothing forces `urwid.display` as part of `sys.path` (like using it as root directory for script call or manual `sys.path` modifications), `absolute_import` will not allow to shadow.

File: urwid/display/_web.js
Comment (penguinolog) [inline urwid/display/_web.js:1]: After extract from python code, static analysis complains about deprecated APIs usage

File: urwid/display/_web.css [hunk 4]
```diff
@@ -0,0 +1,12 @@
+body {    margin: 8px 8px 8px 8px; border: 0;
+    color: black; background-color: silver;
+    font-family: fixed; overflow: hidden; }
```
Comment (penguinolog) [inline urwid/display/_web.css:3]: I have warning "Font property font-family does not have generic default" here

File: urwid/__init__.py [hunk 5]
```diff
@@ -242,3 +244,46 @@
...
+# Backward compatible lazy load with deprecation warnings
+_moved_warn: dict[str, str] = {
+    "escape": "urwid.display.escape",
+    "lcd_display": "urwid.display.lcd",
+    "html_fragment": "urwid.display.html_fragment",
+    "web_display": "urwid.display.web",
+}
+# Backward compatible lazy load without any warnings
+# Before DeprecationWarning need to start PendingDeprecationWarning process.
+_moved_no_warn: dict[str, str] = {
```
Comment (penguinolog) [inline urwid/__init__.py:261]: IMHO, no any warnings/deprecation should be before the next API breaking release (when `DeprecationWarning` will be dropped and `PendingDeprecationWarning` became `DeprecationWarning`).
Then should be long cycle of deprecation (`PendingDeprecationWarning` -> `DeprecationWarning`).
This is too wide used API

File: urwid/display/web.py [hunk 6]
```diff
@@ -624,7 +150,15 @@ def register_palette(self, palette):
                 raise KeyError(f"palette entry '{like_name}' doesn't exist")
             self.palette[name] = self.palette[like_name]
 
-    def register_palette_entry(self, name, foreground, background, mono=None):
+    def register_palette_entry(
```
Comment (penguinolog) [inline urwid/display/web.py:153]: do not break API

File: urwid/display/web.py
Comment (penguinolog) [inline urwid/display/web.py:1]: `str.join` for HTML page generation looks like great candidate for templating usage

**Comment by penguinolog** [comment]
Tested for compatibility with success:
pudb
nomadnet
socli
TextWorld
s-tui
mitmproxy

File: examples/lcd_cf635.py [hunk 7]
```diff
@@ -312,7 +312,7 @@ def build_submenu(ms):
     return build_submenu(menu_structure)
 
 
-screen = urwid.lcd_display.CF635Screen(sys.argv[1])
+screen = urwid.display.lcd_display.CF635Screen(sys.argv[1])
```
Comment (wardi) [inline examples/lcd_cf635.py:315]: this is `display.lcd` now, isn't it?

**Review by wardi — APPROVED** [review]

**Pull request merged**